### PR TITLE
Enhanced email validation regex

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Improves upon #3254

### CODE REVIEW

**Comment by PromoFaux** [comment]
@bash-c / @pvogt09 - please try this branch

**Comment by bash-c** [comment]
I tested and confirm it working

File: advanced/Scripts/webpage.sh [hunk 1]
```diff
@@ -519,7 +519,10 @@ Options:
     if [[ -n "${args[2]}" ]]; then
 
         # Sanitize email address in case of security issues
-        if [[ ! "${args[2]}" =~ ^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$  ]]; then
+        # Regex from https://stackoverflow.com/a/2138832/4065967
+        local regex
+        regex="^[a-z0-9!#\$%&'*+/=?^_\`{|}~-]+(\.[a-z0-9!#$%&'*+/=?^_\`{|}~-]+)*@([a-z0-9]([a-z0-9-]*[a-z0-9])?\.)+[a-z0-9]([a-z0-9-]*[a-z0-9])?\$"
```
Comment (DL6ER) [inline advanced/Scripts/webpage.sh:524]: Looking at this, it seems to allow a lot of dangerous things, like `!` or `=` or even `?` and `$` or `/`. Do we really expect these characters to show up in email addresses?... Especially the exclamation mark may give rise to unpleasant behavior in `bash`!

Comment (DL6ER) [reply advanced/Scripts/webpage.sh:524]: The regex on the SE page you linked goes down all the way into RFC 5322

However, there is also the comment:
> RFC 5322 leaves the domain name part open to implementation-specific choices that won’t work on the Internet today.

Comment (DL6ER) [reply advanced/Scripts/webpage.sh:524]: Thinking again about it, we could maybe even keep the existing Regex and just remove the maximum length limitation of `4`? We do not enforce any restrictions on the former elements of the address, either.

Comment (PromoFaux) [reply advanced/Scripts/webpage.sh:524]: Whoops, you are right. This is why all regex I ever submit needs more pairs of eyes on it...

image

**Comment by bash-c** [comment]
Since it's a security issue and everyone can see this pr. I think it's better to settle it down early.

File: advanced/Scripts/webpage.sh (hunk 1 above)
Comment (DL6ER) [reply advanced/Scripts/webpage.sh:524]: My proposal: Restore the previous regex and simply remove the maximum TLD length of four restriction:

```suggestion
        regex="^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\$"
```

**Pull request merged**